src/Microsoft.SymbolStore.Client/SymbolCache.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.SymbolStore.Client
+{
+    public class SymbolCache
+    {
+        private string _location;
+
+        public SymbolCache(string location)
+        {
+            if (location == null)
+                throw new ArgumentNullException(nameof(location));
+
+            if (string.IsNullOrWhiteSpace(location))
+                throw new ArgumentException($"Argument '{nameof(location)}' cannot be an empty string.");
+
+            _location = location;
+            if (!Directory.Exists(_location))
+                Directory.CreateDirectory(location);
+        }
+
+        #region PDB Functions
+        public string GetPdbFromCache(string pdbName, Guid guid, int age)
```

Comment:
This allows me to finely control the location and layout of all files.  This actually has a very useful property when we end up using the result.  Allowing the user to control where files could go (by accepting a path string, and someone modifying what we return as the index path) would make the rest of the code more difficult..